src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -125,12 +115,4 @@ sealed class BitPat(val value: BigInt, val mask: BigInt, width: Int) extends Sou
       (implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = {
     !(this === that)
   }
-
-  def != (that: UInt): Bool = macro SourceInfoTransform.thatArg
```

Comment:
Might be worth leaving this (or something like this - perhaps as an error) in, because I could see potential for confusion here.

Reply:
Yeah, this is a point of confusion and it's a change to the compatibility mode. Hmm...

Let me think about this. Two questions: (1) what is the precedence problems that the deprecation message of "`=/=` avoids precedence problems" message about? (2) What's with the asymmetry of [how `UInt` and `SInt` implement `!=` as a direct mapping to `=/=`](https://github.com/freechipsproject/chisel3/blob/master/chiselFrontend/src/main/scala/chisel3/Bits.scala#L1124) while others implement it with a macro?

Reply:
For (1), I think this is referring to the problem of getting the Scala compiler to properly choose our `!=`. This is causing me problems trying to write a compatibility layer for, e.g., `SInt`:

```scala
  implicit class SIntCompatibility(a: SInt) {
    import chisel3.internal.sourceinfo.SourceInfo
    final def != (that: SInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = a =/= that
  }
```

The compiler seems to *always* choose the `!=` of `AnyRef` no matter how I mess with this.

Due to the sealed nature of `UInt`, etc. and the fact that the compatibility layer is outside the `chisel3` tree, I'm hitting a wall on how to move `!=` into the compatibility layer.

Reply:
The above is simply an issue with not being able to override using implicit classes. This puts us in a bit of a bind with `!=` though. I think an expedient approach would be to leave `!=` in as perpetually deprecated. That avoids some confusion for new users.